core/arch/arm/sm/sm.c

```diff
@@ -15,7 +16,7 @@
 
 uint32_t sm_from_nsec(struct sm_ctx *ctx)
 {
-	uint32_t *nsec_r0 = (uint32_t *)(&ctx->nsec.r0);
+	struct thread_smc_args *args = (void *)&ctx->nsec.r0;
```

Comment:
I'd prefer casting to `struct thread_smc_args *`

Reply:
ok